src/hotspot/cpu/x86/x86.ad

```diff
@@ -1751,6 +1751,10 @@ bool Matcher::match_rule_supported_vector(int opcode, int vlen, BasicType bt) {
       }
       break;
     case Op_ClearArray:
+      if ((size_in_bits != 512) && !VM_Version::supports_avx512vl()) {
+        return false;
```

Comment:
Please add comment to clarify condition. I am reading it as ClearArray will not be supported for NOT avx512 because we can have vector length 512 bits for not avx512.

Reply:
This is only pertinent to known sized clear arrays which are optimized for AVX-512 targets, we already have such a check as part of matcher predicate, so removing it.